File: app/src/main/java/com/example/sample/sample/myapplication/SpringRecyclerActivity.java
package com.example.sample.sample.myapplication;

import android.app.Activity;
import android.os.Bundle;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.sample.sample.view.CustomAdapter;
import com.example.lib.effect.effect.widget.SpringRecyclerView;
import com.example.sample.myapplication.R;


public class SpringRecyclerActivity extends Activity {
    private static final int DATASET_COUNT = 12;
    protected String[] mDataset;
    protected CustomAdapter mAdapter;
    private SpringRecyclerView mRecyclerView;
    private RecyclerView.LayoutManager mLayoutManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.spring_recyclerview);
        mRecyclerView = findViewById(R.id.recyclerView);
        initDataset();
        mAdapter = new CustomAdapter(mDataset);
        mLayoutManager = new LinearLayoutManager(this);
        mRecyclerView.setLayoutManager(mLayoutManager);
        mRecyclerView.setAdapter(mAdapter);
        mRecyclerView.setHandleTouch(false);
    }

    private void initDataset() {
        mDataset = new String[DATASET_COUNT];
        for (int i = 0; i < DATASET_COUNT; i++) {
            mDataset[i] = "This is element #" + i;
        }
    }
}

File: app/src/main/java/com/example/sample/sample/myapplication/SpringRecyclerActivity.kt
package com.example.sample.sample.myapplication

import android.app.Activity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.lib.effect.effect.widget.SpringRecyclerView
import com.example.sample.myapplication.R
import com.example.sample.sample.view.CustomAdapter

class SpringRecyclerActivity : Activity() {
    //protected val mDataset: Array<String?> = initDataset()
    //private var mLayoutManager: RecyclerView.LayoutManager? = null
    //val mAdapter = CustomAdapter(initDataset())
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.spring_recyclerview)
        var mRecyclerView : SpringRecyclerView = findViewById(R.id.recyclerView)
        //initDataset()
        //mAdapter = CustomAdapter(mDataset)

        //mLayoutManager = LinearLayoutManager(this)
        mRecyclerView.setLayoutManager(LinearLayoutManager(this))
        mRecyclerView.setAdapter(CustomAdapter(Array<String?>(DATASET_COUNT) { "This is element $it" }))
        mRecyclerView.setHandleTouch(false)
    }

    companion object {
        private const val DATASET_COUNT = 12
    }
}